src/node/util.ts

```diff
@@ -485,6 +485,15 @@ export function isNodeJSErrnoException(error: unknown): error is NodeJS.ErrnoExc
   return error !== undefined && (error as NodeJS.ErrnoException).code !== undefined
 }
 
+/**
+ * A helper function which returns a boolean indicating whether
+ * the given address is AddressInfo and has .address
+ * and a .port property.
+ */
+export function isAddressInfo(address: unknown): address is net.AddressInfo {
+  return address !== null && typeof address !== "string" && (address as net.AddressInfo).port !== undefined
+}
```

Comment:
We might want to also check `.address` assuming we leave this as `unknown`, or alternatively change the type to something like `string | net.AddressInfo` (otherwise objects like `{ port: 10 }` can be passed in and return true).

Reply:
Ah, good call on location. I lazily copied a function above and just left it in the same location. 

Good call on checking for `.address` too 👍